c10/util/Float8_e8m0fnu.h

```diff
@@ -0,0 +1,119 @@
+#pragma once
+
+/// Defines the Float8_e8m0fnu type (8-bit floating-point) including
+/// conversions to standard C types
+/// Binary configuration :
+/// eeeeeeee
+/// no sign bits
+/// 8 exponent bits
+/// no mantissa bits
+///
+/// This is the E8M0 dtype from the OCP MX format spec
+/// (https://www.opencompute.org/documents/ocp-microscaling-formats-mx-v1-0-spec-final-pdf,
+/// Section 5.4.1)
+
+#include <c10/macros/Export.h>
+#include <c10/macros/Macros.h>
+#include <c10/util/floating_point_utils.h>
+#include <type_traits>
+
+#if defined(__cplusplus)
+#include <cstdint>
+#elif !defined(__OPENCL_VERSION__)
```

Comment:
When do we build with OPENCL ?!

Reply:
This code was added to Half.h in https://github.com/pytorch/pytorch/commit/5ed9dfad982a4e20a51fdf30d3620e76b87ecff4 and looks like all the float8 dtypes copy-pastaed it.  TBD if this is still needed.  Tracking in https://github.com/pytorch/pytorch/issues/146647